sdk/program/src/alt_bn128/compression.rs

```diff
@@ -86,7 +87,7 @@ mod target_arch {
         let decompressed_g1 = G1::deserialize_with_mode(
             convert_endianness::<32, 32>(&g1_bytes).as_slice(),
             Compress::Yes,
-            Validate::No,
+            Validate::Yes,
```

Comment:
Point validation was skipped because is it very expensive, a multiple of the compute required for the actual decompression, and it is already being performed in the altbn syscall.
Is there a specific edge case you have in mind that could still occur?
See:
https://github.com/anza-xyz/agave/blob/2c125005133b2459b62965a0fc20c49c82392350/sdk/program/src/alt_bn128/mod.rs#L117
https://github.com/anza-xyz/agave/blob/2c125005133b2459b62965a0fc20c49c82392350/sdk/program/src/alt_bn128/mod.rs#L143

Reply:
I'm ok skipping validation. My fear is just that diff libs will behave differently, we can doc the expected behavior in simd. @samkim-crypto opinions?
I've updated the PR with Validate::No.

@ananas-block before we had `G2::deserialize_compressed`, i.e. g2-compressed was the only one doing validation. Was that intended? I'd leave them all explicit, so it's a bit more clear.

Reply:
Yes, I think skipping validation is fine to for me as in the original implementation..